src/shared/components/query/QueryStoreUtils.ts

```diff
@@ -119,34 +120,88 @@ export function profileAvailability(molecularProfiles:MolecularProfile[]) {
+    let filteredCnaSamples: { [studyId: string]: { [sampleId: string]: boolean } } = {};
+    let filteredMutationCnaSamples: { [studyId: string]: { [sampleId: string]: boolean } } = {};
+    let filteredallSamples: { [studyId: string]: { [sampleId: string]: boolean } } = {};
+
+    sampleLists
+        .filter(sampleList => _.includes(['all_cases_in_study', 'all_cases_with_mutation_and_cna_data', 'all_cases_with_mutation_data', 'all_cases_with_cna_data'], sampleList.category))
+        .forEach(sampleList => {
+            let samples: { [sampleId: string]: boolean } =
+                _.reduce(sampleList.sampleIds, (acc: { [sampleId: string]: boolean }, next) => {
+                    acc[next] = true;
+                    return acc;
+                }, {});
+
+            switch (sampleList.category) {
+                case "all_cases_with_mutation_and_cna_data":
+                    mutationCnaSamples[sampleList.studyId] = samples;
+                    break;
+                case "all_cases_with_mutation_data":
+                    mutationSamples[sampleList.studyId] = samples;
+                    break;
+                case "all_cases_with_cna_data":
+                    cnaSamples[sampleList.studyId] = samples;
+                    break;
+                case "all_cases_in_study":
+                    allSamples[sampleList.studyId] = samples;
+                    break;
+            }
+        });
+
+    let selectedVirtualStudySet: { [id: string]: VirtualStudy } =_.reduce(selectedVirtualStudies, (acc: { [id: string]: VirtualStudy }, next) => {
+            acc[next.id] = next; return acc
+    }, {});
+    let selectedPhysicalStudyIds = selectedStudies.filter(id => !_.has(selectedVirtualStudySet, id));
+
+    //add all samples from selected physical studies
+    _.forEach(selectedPhysicalStudyIds,studyId=>{
+        filteredMutationSamples[studyId] = mutationSamples[studyId] || {};
+        filteredCnaSamples[studyId] = cnaSamples[studyId] || {};
+        filteredMutationCnaSamples[studyId] = mutationCnaSamples[studyId] || {};
+        filteredallSamples[studyId] = allSamples[studyId] || {};
+    });
+
+    _.forEach(selectedVirtualStudySet,virtualStudy=>{
+        _.forEach(virtualStudy.data.studies,study=>{
+            
+            // check if the study in this virtual study is already in the selected studies list
+            // and only add the samples if its not already present
+            if(!_.includes(selectedPhysicalStudyIds,study.id)){
+                filteredMutationSamples[study.id] = filteredMutationSamples[study.id] || {};
+                filteredCnaSamples[study.id] = filteredCnaSamples[study.id] || {};
+                filteredMutationCnaSamples[study.id] = filteredMutationCnaSamples[study.id] || {};
+                filteredallSamples[study.id] = filteredallSamples[study.id] || {};
+
+                _.forEach(study.samples,sample=>{
```

Comment:
this is kind of confusing, but it's not sample, it's sampleId.  We need to be really careful about what we name things.  I know it comes from VirtualStudy model, but that should be changed.

Reply:
I'll change it